src/tools/eraser-tool.ts

```diff
@@ -126,15 +127,54 @@ function handleMouseClick(worldClickPoint: Point) {
       const rectangle = closestEntity.entity as RectangleEntity;
       const rectangleShape = rectangle.getShape() as Box;
       const segments = rectangleShape.toSegments();
-      const segmentEntities = segments.map(segment => new LineEntity(segment));
-      // Replace rectangle with its line segments in the entities array
-      deleteEntity(rectangle);
-      addEntity(...segmentEntities);
-
-      segmentEntities.forEach(segmentEntity =>
-        eraseLineSegment(segmentEntity, clickedPointOnShape, intersections),
-      );
+      
+      // Find the closest segment to the clicked point
+      const closestSegment = findClosestEntity(worldClickPoint, segments.map(segment => new LineEntity(segment)));
+      
+      if (closestSegment) {
+        // Remove the rectangle
+        deleteEntity(rectangle);
+        
+        // Create line entities for all sides
+        segments.forEach(segment => {
+          const segmentIntersections = intersections.filter(point => segment.contains(point));
+          
+          if (segmentIntersections.length > 0) {
+            // If the segment has intersections, split it
+            const sortedPoints = [segment.start, ...segmentIntersections, segment.end]
+              .sort((a, b) => a.distanceTo(segment.start)[0] - b.distanceTo(segment.start)[0]);
+            
+            for (let i = 0; i < sortedPoints.length - 1; i++) {
+              const subSegment = new Segment(sortedPoints[i], sortedPoints[i + 1]);
+              if (!isPointOnSegment(worldClickPoint, subSegment)) {
+                const lineEntity = new LineEntity(subSegment);
+                lineEntity.lineColor = rectangle.lineColor;
+                lineEntity.lineWidth = rectangle.lineWidth;
+                addEntity(lineEntity);
+              }
+            }
+          } else if (!areLineSegmentsEqual(segment, (closestSegment.entity as LineEntity).getShape() as Segment)) {
+            // If the segment is not the clicked one and has no intersections, add it as is
+            const lineEntity = new LineEntity(segment);
+            lineEntity.lineColor = rectangle.lineColor;
+            lineEntity.lineWidth = rectangle.lineWidth;
+            addEntity(lineEntity);
+          }
+        });
+      }
```

Comment:
I seem to have to click twice to delete a line. See this video:

https://github.com/user-attachments/assets/3446c69e-1004-4e7c-8aa0-d520b10e06cd

Also you seem to be doing a lot of unneeded work, since there alre already some helper functions available:

```suggestion
      const segmentEntities = segments.map(segment => new LineEntity(segment));

      // Find the closest segment to the clicked point
      const closestSegmentInfo = findClosestEntity(
        worldClickPoint,
        segmentEntities,
      );

      // Remove the rectangle
      deleteEntity(rectangle);

      // Add 4 lines instead of the rectangle
      addEntity(...segmentEntities);

      // Remove (a segment) of the 4th line closest to the cursor
      const lineIntersections = getAllIntersectionPoints(
        closestSegmentInfo.entity,
        getEntities(),
      );
      eraseLineSegment(
        closestSegmentInfo.entity as LineEntity,
        clickedPointOnShape,
        lineIntersections,
      );
```